### ISSUE TITLE
Rewrite to ES6

### ISSUE THREAD

**Issue by davixyz** [amzn/style-dictionary#153]
Had a talk with @dbanksdesign and basically we could start migrating the library little by little to use ES6 features. Of course this needs some infrastructure work (that I would love to do) and also to take some decisions about how do we do this. Opening this issue to start the conversation for this.

### Some Thoughts
- Some of the features are already available in node 6 and node 8, probably the most important ones we would want like template literals, spreads, destructuring etc
- Replacing and using arrow functions everywhere might be tricky because of how style dictionary API has been designed with `this` in mind
- A build system will be required to ship the code for older versions

### Some build systems
- **Webpack**: https://webpack.js.org/guides/author-libraries/
- **Rollup**: https://rollupjs.org/guide/en

**Comment by vvscode** [comment]
There is https://github.com/5to6/5to6

I'm ready to do it

**Comment by chazzmoney** [comment]
Thats cool! I didn't know that was even a thing that existed.

If you are ready, go for it. Otherwise I can take a look next week.

**Comment by chazzmoney** [comment]
Probably best done in conjunction with #154.

I think we should wait on this until we merge #169 before doing this.

**Comment by vvscode** [comment]
So I'm waiting for merge https://github.com/amzn/style-dictionary/pull/169 to make this one and https://github.com/amzn/style-dictionary/issues/154

To which branch?

**Comment by dbanksdesign** [comment]
develop-3.0

**Comment by vvscode** [comment]
@dbanksdesign @chazzmoney  are you going to merge `develop` to `develop-3.0` ?

**Comment by chazzmoney** [comment]
Yes, sorry for the delay. Going to try to get #158 and #152 merged into develop in the next couple days and then merge develop into develop-3.0

**Comment by dbanksdesign** [comment]
develop-3.0 is now up to date with develop

**Comment by chazzmoney** [comment]
Promise to get #152 merged soon - almost there. Thanks for your patience @vvscode.

**Comment by dbanksdesign** [comment]
#500 is the beginnings of this

**Comment by dbanksdesign** [comment]
@chazzmoney you think we can close this?

**Comment by davixyz** [comment]
If this is finished, we should close it! Hello everyone!

**Comment by dbanksdesign** [comment]
This is done (for the most part)! Closing!
